CIP50 Optional replay protection (#298)
celo-org/celo-proposals: CIPs/cip-0050.md

---
cip: 50
title: Optional Replay Protection
author: Gastón Ponti (@gastonponti)
discussions-to: <URL>
status: Draft
type: <Standards Track, Meta or Informational>
category (*only required for Standards Track): <Ring 0, 1, 2, 3>
created: 2021-11-18
requires (*optional): <CIP number(s)>
replaces (*optional): <CIP number(s)> (changes part of the cip35(?))
license: Apache 2.0
---

This is the suggested template for new CIPs.

Note that a CIP number will be assigned by an editor. When opening a pull request to submit your CIP, please
use an abbreviated title in the filename, `cip-draft_title_abbrev.md`.

## Simple Summary
Revert one of the changes added in the [CIP-35](https://github.com/celo-org/celo-proposals/blob/master/CIPS/cip-0035.md) which made the replay protection (see [EIP-155](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md)) mandatory

## Abstract
A short (~200 word) description of the technical issue being addressed.

## Motivation

> Making the replay protection mandatory breaks our compatibility with Ethereum. EIPs as for example the 1820 depend exclusive in not having that limitation. If our biggest concern was to avoid errors or confusion between chains, that should be check by our tooling or rpc api calls, not a evm limitation.

Reviewer: ```suggestion
Making the replay protection mandatory breaks our compatibility with Ethereum. EIPs such as [EIP-1820](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1820.md) depend exclusively on not having that limitation. If our biggest concern was to avoid errors or confusion between chains, that should be checked by our tooling or rpc api calls, but it should not be an evm limitation.
```
